packages/validator/src/services/doppelgangerService.ts

```diff
@@ -0,0 +1,266 @@
+import {Epoch, ValidatorIndex} from "@chainsafe/lodestar-types";
+import {Api} from "@chainsafe/lodestar-api";
+import {ILogger, sleep} from "@chainsafe/lodestar-utils";
+import {computeStartSlotAtEpoch} from "@chainsafe/lodestar-beacon-state-transition";
+import {ProcessShutdownCallback, PubkeyHex} from "../types.js";
+import {IClock} from "../util/index.js";
+import {Metrics} from "../metrics.js";
+import {IndicesService} from "./indices.js";
+
+// The number of epochs that must be checked before we assume that there are
+// no other duplicate validators on the network
+const DEFAULT_REMAINING_DETECTION_EPOCHS = 1;
+const REMAINING_EPOCHS_IF_DOPPLEGANGER = Infinity;
+
+export type LivenessResponseData = {
+  index: ValidatorIndex;
+  epoch: Epoch;
+  isLive: boolean;
+};
+
+export type DoppelgangerState = {
+  nextEpochToCheck: Epoch;
+  remainingEpochs: Epoch;
+};
+
+export enum DoppelgangerStatus {
+  // This pubkey is known to the doppelganger service and has been verified safe
+  VerifiedSafe = "VerifiedSafe",
+  // This pubkey is known to the doppelganger service but has not been verified safe
+  Unverified = "Unverified",
+  // This pubkey is unknown to the doppelganger service
+  Unknown = "Unknown",
+  // This pubkey has been detected to be active on the network
+  DoppelgangerDetected = "DoppelgangerDetected",
+}
+
+/** Map a DoppelgangerStatus to an integer for rendering in Grafana */
+export const doppelgangerStatusMetrics: {[K in DoppelgangerStatus]: number} = {
```

Comment:
I see this is no longer used. If it won't be anymore maybe remove?

Reply:
As talked with @dadepo I'm happy to add it back latter if there is demand of per-validator metrics